File: leetcode/golang/19.删除链表的倒数第n个节点.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func removeNthFromEnd(head *ListNode, n int) *ListNode {
    var start *ListNode = &ListNode{
        Val: 0,
        Next: head,
    };

    first := start;
    second := start;

    for n > 0 {
        second = second.Next;
        n -= 1;
    }

    for second.Next != nil {
        first = first.Next;
        second = second.Next;
    }
    first.Next = first.Next.Next;

    return start.Next;
}

File: leetcode/python/19.删除链表的倒数第n个节点.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        start = ListNode()
        start.next = head

        first = start
        second = start
        for i in range(n):
            second = second.next
        
        while second.next :
            first = first.next
            second = second.next
        
        first.next = first.next.next
        return start.next

